### DISCUSSION TITLE
Issue on prompt or image? Help me figure it out.

### DISCUSSION THREAD

**Discussion by oldhand7** [lllyasviel/Fooocus#2744] [Q&A]
I 've uploaded 4 images of poor clothes women on the image prompt and set to "swap" method and set the text prompt to wedding guest. The result is not expected. It seems the model also picks up on what clothes the user is wearing in the photo, however users may upload poor photos of themselves which may not match the clothes they want. Perhaps there's something we can do here.
Appreciate your help on this matter. Do I have to change image or update text prompt? 

Sincerely,

**Comment by ILikeToasters** [comment]
I  am not sure I completely understand. You mention "swap" method which would be the FaceSwap. When using Faceswap it will often pick up on what the person is wearing near their faces but can often be overcome with a good text prompt. Give information about the clothing for example in your text prompt and that should make a difference. I find that if you use multiple face images with different clothing or background that effect is much less. But if you are using 4 images that are similar clothing of the same person that will have a heavier influence.

**Comment by mashb1t** [comment]
Please provide examples of what you're trying to achieve and what your input is so you can be helped effectively.

**Comment by oldhand7** [comment]
![Untitled](https://github.com/lllyasviel/Fooocus/assets/91205658/608365d2-aed2-4c9d-9005-78d934cb1440)

**Comment by oldhand7** [comment]
I input 4 images with dark poor clothes for image prompt and textprompt as travel.
![image](https://github.com/lllyasviel/Fooocus/assets/91205658/6b83e150-e79d-49e9-85f3-abc9e966708e)
the result is unexpected. The output photo usually contains dark, black clothes which is not suitable for the travel.
I expected to bright beautiful color clothes
I guess we can do something to solve this problem.
Hope to hear from experts.
Thank you.

**Reply by ILikeToasters** [reply]
Because the purpose of the image prompt is to use everything in those images to influence the outcome. If you want to just use the face enable the advanced check box at the bottom and use the faceswap feature. 
<img width="392" alt="Screenshot 2024-04-11 200413" src="https://github.com/lllyasviel/Fooocus/assets/4357657/3d84be7a-e1f3-4521-b623-53f689fa142a">

**Comment by oldhand7** [comment]
Thank you. but  actually I set to the faceswap mode and generated photo face is similar to image prompt but the output clothes are not expected. I input poor dark clothes women but dark output clothes are not suitable for trip, travel, party for women. Furthermore, I need to generate various kinds of clothes of women such as yellow, white, blue , green but only black, white clothes are  generated.
Appreciate it if you can help us.

**Reply by ILikeToasters** [reply]
You have not said what you had for a prompt exactly. Did you describe the clothing and colors? If not then those images will still influence the results even with faceswap. But if you prompt for different things specifically it should make a difference. For example I just used my previous image to do faceswap and got different clothing because I specified something different. As you can see no white shirt.
![tianaxx (8)](https://github.com/lllyasviel/Fooocus/assets/4357657/8697cb84-758a-4750-8e07-c3d4ae7ff784)

**Comment by oldhand7** [comment]
I described default prompt as "woman with various colors , styles for " 

and input additional various kinds of situation for prompt like "cocktail party", "builing", "working", "cooking". Then we can add them to default prompt 



One thing I am worried about is that when user wear a dark clothes, the result clothes are often dark black but when user wears a bright color clothes for input image prompt, the result is also bright color clothes for result regardless of input text prompt.

I think I can update the text prompt so that we can avoid this happening.

Thank you for your help.
Feel free to let me know if you have any good idea on it.

**Reply by Rogala** [reply]
The image was of poor quality, and it was fast.

Original image
![0](https://github.com/lllyasviel/Fooocus/assets/7437582/728435f9-e7c8-48c0-9100-89979163393e)



`{
"prompt": "woman is wearing a white evening dress",
"negative_prompt": "",
"prompt_expansion": "woman is wearing a white evening dress, intricate, elegant, highly detailed, dramatic light, sharp focus, clear excellent composition, dynamic, cinematic, ambient, pristine, beautiful, creative, innocent, extremely inspirational, color, shiny, attractive, decorated, very coherent, cute, elaborate, artistic, lovely, surreal, pretty, inspired, fancy, enhanced, complex background, perfect",
"styles": "['Fooocus V2']",
"performance": "Lightning",
"resolution": "(960, 1024)",
"guidance_scale": 1.0,
"sharpness": 0.0,
"adm_guidance": "(1.0, 1.0, 0.0)",
"base_model": "juggernautXL_v8Rundiffusion.safetensors",
"refiner_model": "None",
"refiner_switch": 1.0,
"adaptive_cfg": 1.0,
"sampler": "euler",
"scheduler": "sgm_uniform",
"seed": "5611857620814568828",
"lora_combined_1": "sd_xl_offset_example-lora_1.0.safetensors : 0.1",
"lora_combined_2": "MJ52.safetensors : 0.32",
"lora_combined_3": "sdxl_lightning_4step_lora.safetensors : 1.0",
"metadata_scheme": false,
"version": "Fooocus v2.3.1"
}`
![2024-04-12_08-46-04_6538](https://github.com/lllyasviel/Fooocus/assets/7437582/f35fc033-cd3d-4f1d-8636-cd3abae8dece)

woman is wearing a red dress
![2024-04-12_08-40-06_3485](https://github.com/lllyasviel/Fooocus/assets/7437582/aba71d41-5056-442e-ab43-d90ba8b5b2cf)
woman is wearing a yellow T-shirt and green shorts
![2024-04-12_08-38-55_4099](https://github.com/lllyasviel/Fooocus/assets/7437582/2466e19c-228d-4183-bff8-d56f356a9820)
woman is wearing a blue jeans and a white sweater
![2024-04-12_08-29-43_3329](https://github.com/lllyasviel/Fooocus/assets/7437582/36de8c35-e5ce-44b6-bd7e-7b81b54cd4bf)
woman is wearing a blue suit
![2024-04-12_08-27-21_4898](https://github.com/lllyasviel/Fooocus/assets/7437582/332f382f-2862-481e-874c-ed4982cfdd64)

Try to express your requests explicitly:
1 The name of the object
2 The action that happens to it
3 What it looks like
4 What additional objects are around it and their action

The further away from the beginning of the promt the text is, the less weight it has.

**Comment by oldhand7** [comment]
Thanks Rogala.
My Goal: I upload 4 dark poor clothes that woman doesn't like. I text input prompt as "garden party" We expect the result images are bright clothes but they are often dark black clothes as input images. I used "faceswap" mode to keep the face of input image.

Please let me know how to configure or set the text prompt properly to aviod this? Or should someone update the codebase?
Or should I just upload only face image prompts? Or should I crop input images into face section?

Thank you for helping me.

Best.

**Reply by Rogala** [reply]
If you think the AI knows how people dress in this or that situation, it doesn't, write in the request specifically what clothes you want to get and what color they will be.

And in **negative_prompt:** write what colors you don't want to see.

You just understand the AI is not a magic wand, thought and ready, it waits for a specific task, a little detail it takes from models and lore independently, but to direct what to take should be you yourself.

**Comment by oldhand7** [comment]
I understand. If I upload only person's face, I can solve the problem, right?

**Reply by Rogala** [reply]
If you don't describe what she'll be wearing and where she'll be, the result could be anything with your face.

![194f46948b5e6df001f3cd5b3e400a99](https://github.com/lllyasviel/Fooocus/assets/7437582/5abf14ce-81e7-4df5-9c41-fca2c7af4d2c)

```
{
"prompt": "a girl in the garden in a light floral dress",
"negative_prompt": "",
"prompt_expansion": "",
"styles": "[]",
"performance": "Lightning",
"resolution": "(1088, 960)",
"guidance_scale": 1.0,
"sharpness": 0.0,
"adm_guidance": "(1.0, 1.0, 0.0)",
"base_model": "juggernautXL_v8Rundiffusion.safetensors",
"refiner_model": "None",
"refiner_switch": 1.0,
"adaptive_cfg": 1.0,
"sampler": "euler",
"scheduler": "sgm_uniform",
"seed": "5692495374808126241",
"lora_combined_1": "sd_xl_offset_example-lora_1.0.safetensors : 0.1",
"lora_combined_2": "MJ52.safetensors : 0.45",
"lora_combined_3": "clear.safetensors : 2.0",
"lora_combined_4": "sdxl_lightning_4step_lora.safetensors : 1.0",
"metadata_scheme": false,
"version": "Fooocus v2.3.1"
}
```
![2024-04-12_22-07-05_4655](https://github.com/lllyasviel/Fooocus/assets/7437582/21cfa408-f2cd-4e05-b684-e2dffda607eb)
